app/models/form_profiles/va_686c674.rb

```diff
@@ -34,7 +34,13 @@ def metadata
   private
 
   def prefill_form_address
-    mailing_address = VAProfileRedis::ContactInformation.for_user(user).mailing_address if user.vet360_id.present?
+    replace_pciu = Flipper.enabled?(:remove_pciu, user)
```

Comment:
Can some of this be extracted to reduce duplication?  Something like...
```ruby
module ContactInformationVersioning
  extend ActiveSupport::Concern

  # Determine whether to use the new VAProfile V2 system
  def use_v2_service?(user)
    Flipper.enabled?(:remove_pciu, user)
  end

  # Retrieve the user's contact information
  def contact_information_from(user)
    if use_v2_service?(user)
      VAProfileRedis::V2::ContactInformation.for_user(user)
    else
      VAProfileRedis::ContactInformation.for_user(user)
    end
  end

  # Retrieve mailing address based on conditions
  def mailing_address_from(user)
    redis_prefill = contact_information_from(user)
    use_v2_service = use_v2_service?(user)
    # Only assign mailing address if the feature flag is enabled or the user has a vet360_id
    redis_prefill&.mailing_address if use_v2_service || user.vet360_id.present?
  end
end
```

Just a suggestion.

Reply:
I would agree to split this out if there were not steps after this to remove all the Contact Info V1 references.  I believe the new version will be replaced soon after being pushed to prod. This would require more files to be removed when we remove PCIU and Contact Information in the later steps of the [PCIU migration epic](https://github.com/orgs/department-of-veterans-affairs/projects/1335/views/6?pane=issue&itemId=75083733&issue=department-of-veterans-affairs%7Cva.gov-team%7C73776). Both Contact Information V1 and PCIU need to removed after this integration has been tested.

Reply:
I was thinking along the same lines and thought having a module/concern in place could make future upgrades more convenient. But, you're probably right - keeping it as-is for now might simplify clean up.